### PR TITLE
[Core] Adding FullName method to ModelPart

### LINKED ISSUE #6056: [Core] (Sub)ModelPart Name()
When the name of a model part is retrieved, the method `Name()` always returns the value in `mName`. In case the model part of interest is a submodelpart, it returns the submodelpart name without the parent model part name prepending it.

After #6024 this might turn some warnings to errors. So far I've found one case in the FluidDynamicsApplication, but there could be more in other applications.

Having said this, I see two options:
- We keep the current behaviour of `Name()` and we implement a `FullName()` method.
- We modify the current behaviour of `Name()` to recursively check if the model part of interest is a submodelpart to prepend `mName` with the father model part model part name.

Personally, I'd rather go for the second one. However I haven't checked if there is any technical limitation.

### PR DESCRIPTION
**Pull request by loumalouomega**
Fixes #6056

### CODE REVIEW

File: kratos/includes/model_part.h
Comment (philbucher) [inline kratos/includes/model_part.h]: why the non-const method? I don't think this should be there

**Review by philbucher — COMMENTED** [review]
can you please add a test and also py-exposure?
Thx

File: kratos/tests/cpp_tests/sources/test_model_part.cpp
Comment (rubenzorrilla) [inline kratos/tests/cpp_tests/sources/test_model_part.cpp:140]: Would you mind to test an extra level (subsubmodelpart) to ensure that the recursivity always work?

Comment (loumalouomega) [reply kratos/tests/cpp_tests/sources/test_model_part.cpp:140]: Done

File: kratos/includes/model_part.h
Comment (philbucher) [inline kratos/includes/model_part.h]:
```suggestion
     * @brief This method returns the full name of the model part (including the parents model parts)
```

**Review by philbucher — DISMISSED** [review]
this is what we discussed, I don't think we need to again discuss this :)

Can you please wait until tmr with merging, just in case?

Thanks

**Comment by loumalouomega** [comment]
> 
> 
> this is what we discussed, I don't think we need to again discuss this :)
> 
> Can you please wait until tmr with merging, just in case?
> 
> Thanks

I applied you suggestion, I will reapproval

**Review by philbucher — APPROVED** [review]

**Comment by loumalouomega** [comment]
I will wait until tomorrow, in any case feel free to merge anytime

**Pull request merged**
